### PR TITLE
[REF] web: remove dropdown-item-selected event 

### PR DESCRIPTION
**Pull request by sdegueldre**
This is in prepartion for owl2 where event handlers are no longer
supported on components

### CODE REVIEW

File: addons/web/static/src/search/comparison_menu/comparison_menu.xml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <templates xml:space="preserve">
 
     <t t-name="web.ComparisonMenu" owl="1">
-        <Dropdown class="o_comparison_menu btn-group" togglerClass="'btn btn-light'" t-on-dropdown-item-selected="onComparisonSelected">
+        <Dropdown class="o_comparison_menu btn-group" togglerClass="'btn btn-light'" onItemSelected="payload => onComparisonSelected(payload)">
```
Comment (Polymorphe57) [inline addons/web/static/src/search/comparison_menu/comparison_menu.xml:5]: why keep payload here? Elsewhere, you use a better name

Comment (Polymorphe57) [reply addons/web/static/src/search/comparison_menu/comparison_menu.xml:5]: hum not everywhere. Could you make this uniform?

File: addons/web/static/src/views/pivot/pivot_group_by_menu.js [hunk 2]
```diff
@@ -46,14 +46,16 @@ export class PivotGroupByMenu extends GroupByMenu {
      * @override
      * @param {CustomEvent} ev
```
Comment (Polymorphe57) [inline addons/web/static/src/views/pivot/pivot_group_by_menu.js:47]: change doc

Comment (Polymorphe57) [reply addons/web/static/src/views/pivot/pivot_group_by_menu.js:47]: everywhere

File: addons/web/static/src/views/pivot/pivot_group_by_menu.js [hunk 3]
```diff
@@ -44,16 +44,20 @@ export class PivotGroupByMenu extends GroupByMenu {
...
-        const { itemId, optionId } = ev.detail.payload;
-        const item = this.items.find(({ id }) => id === itemId);
-        // Augment payload data
-        Object.assign(ev.detail.payload, {
-            fieldName: item.fieldName,
-            interval: optionId,
-            groupId: this.props.cell.groupId,
-        });
+    onGroupBySelected({ itemId, optionId }) {
+        if (this.props.onItemSelected) {
```
Comment (aab-odoo) [inline addons/web/static/src/views/pivot/pivot_group_by_menu.js:52]: isn't the prop mandatory (a fortiori for the PivotGroupByMenu)?

**Review by aab-odoo — APPROVED** [review]
Seems good. Maybe worth waiting for @brboi to take a look since he's the dropdown expert.

File: addons/web/static/src/views/pivot/pivot_group_by_menu.js (hunk 3 above)
Comment (sdegueldre) [reply addons/web/static/src/views/pivot/pivot_group_by_menu.js]: Good point. I've made it mandatory through props validation.

File: addons/web/static/src/webclient/user_menu/user_menu.js [hunk 4]
```diff
@@ -1,28 +1,14 @@
 /** @odoo-module **/
 
 import { browser } from "@web/core/browser/browser";
-import { DropdownItem } from "@web/core/dropdown/dropdown_item";
+import { MenuItem } from "@web/webclient/navbar/navbar";
```
Comment (brboi) [inline addons/web/static/src/webclient/user_menu/user_menu.js:4]: It feels weird having to import a MenuItem from navbar/navbar for a UserMenu that does not look - at first glance - directly related to the Navbar (from a file structure perspective).

**Review by brboi — APPROVED** [review]
Love it.
Nothing to say besides a small question we could consider, even if it is totally mergeable IMO.

File: addons/web/static/src/webclient/user_menu/user_menu.js (hunk 4 above)
Comment (brboi) [reply addons/web/static/src/webclient/user_menu/user_menu.js:4]: (just wondering if we should rethink a bit the way files are structured under src/webclient)

**Review by brboi — COMMENTED** [review]
Still love it <3

Documentation should be updated at least for the new dropdown item prop

**Comment by ged-odoo** [comment]
robodoo r+

**Comment by aab-odoo** [comment]
robodoo rebase-ff

**Pull request closed without merging**